pkgs/development/libraries/physics/geant4/default.nix

```diff
@@ -44,6 +53,16 @@ stdenv.mkDerivation rec {
     sha256 = "0vznm3pjlbihjy1wsxc4gj229k0dzc283wvil2xghyl08vwdpnpc";
   };
 
+  boost_python_lib = "python${builtins.replaceStrings ["."] [""] python3.pythonVersion}";
```

Comment:
This might be a widely known thing.  But it's the first time I've encountered these particular "make boost work with python" incantations.  Can you add a comment explaining?  Or is this just something that everyone who needs to know already knows / explained elsewhere?

Specifically, I'm curious because this is different from the previous version in the g4py/default.nix file:

```
  postPatch = ''
    cd environments/g4py
  '';

  preConfigure = ''
    # Fix for boost 1.67+
    substituteInPlace CMakeLists.txt \
    --replace "find_package(Boost)" "find_package(Boost 1.40 REQUIRED COMPONENTS python${builtins.replaceStrings ["."] [""] python.pythonVersion})"
    for f in `find . -name CMakeLists.txt`; do
      substituteInPlace "$f" \
        --replace "boost_python" "\''${Boost_LIBRARIES}"
    done
  '';
```

Reply:
Yes, this is a known breaking change in boost. With respect to upstream this is https://bugzilla-geant4.kek.jp/show_bug.cgi?id=2059